File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_salesorderpagedresponselinksmodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_datacontracts_datasets_salesorderpagedresponsehrefmodel as shared_codat_datacontracts_datasets_salesorderpagedresponsehrefmodel


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsSalesOrderPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_salesorderpagedresponsehrefmodel.CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_salesorderpagedresponsehrefmodel.CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_salesorderpagedresponsehrefmodel.CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_salesorderpagedresponsehrefmodel.CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetssalesorderpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel } from "./codatdatacontractsdatasetssalesorderpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsSalesOrderPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsSalesOrderPagedResponseHrefModel;
}